applications/OptimizationApplication/python_scripts/algorithms/standardized_NLOPT_constraint.py

```diff
@@ -0,0 +1,180 @@
+
+        parameters.ValidateAndAssignDefaults(default_parameters)
+
+        response = optimization_problem.GetResponse(parameters["response_name"].GetString())
+
+        super().__init__(master_control, response)
+
+        if required_buffer_size < 2:
+            raise RuntimeError(f"Standardized constraint requires 2 as minimum buffer size. [ response name = {self.GetReponse().GetName()} ]")
+
+        component_data_view = ComponentDataView(response, optimization_problem)
+        component_data_view.SetDataBuffer(required_buffer_size)
+
+        self.__optimization_problem = optimization_problem
+        self.__buffered_data = component_data_view.GetBufferedData()
+        self.__unbuffered_data = component_data_view.GetUnBufferedData()
+
+        if parameters["ref_value"].IsDouble():
+            self.__ref_type = "specified_value"
+            self.__reference_value = parameters["ref_value"].GetDouble()
+        elif parameters["ref_value"].IsString() and parameters["ref_value"].GetString() == "initial_value":
+            self.__ref_type = "initial_value"
+            self.__reference_value = None
+        else:
+            raise RuntimeError(f"Provided \"reference_type\" = {self.__ref_type} is not supported for constraint response functions. Followings are supported options: \n\tinitial_value\n\tfloat value")
+
+        self.__constraint_type = parameters["type"].GetString()
+        if self.__constraint_type in ["=", "<"]:
+            self.__scaling = 1.0
+        elif self.__constraint_type in [">"]:
+            self.__scaling = -1.0
+            if not self.__reference_value is None: self.__reference_value *= -1
+        else:
+            raise RuntimeError(f"Provided \"type\" = {self.__constraint_type} is not supported in constraint response functions. Followings are supported options: \n\t=\n\t<\n\t>")
+
+        self.__zero_threshold = sys.float_info.epsilon
+
+    def IsEqualityType(self) -> str:
+        return self.__constraint_type == "="
+
+    def GetReferenceValue(self) -> float:
+        if self.__reference_value is not None:
+            return self.__reference_value
+        else:
+            if self.__unbuffered_data.HasValue("initial_value"):
+                self.__reference_value = self.__unbuffered_data["initial_value"]
+                return self.__reference_value
+            else:
+                raise RuntimeError(f"Response value for {self.GetReponse().GetName()} is not calculated yet.")
+
+    def Initialize(self):
+        super().Initialize()
+
+    def CalculateStandardizedValueAndGradients(self, control_field: numpy.ndarray, gradient_field: numpy.ndarray, save_value: bool = True) -> float:
+
+        # update the master control
+        update_status = self.GetMasterControl().Update(control_field)
+        master_control_updated = False
+        for is_updated in update_status.values():
+            if is_updated:
+                master_control_updated = True
+                break
```

Comment:
```suggestion
        master_control_updated = any(update_status)
```

might need to play with the syntax, but `any` should work here